fileconnector/fileconnector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FileCopy.java

```diff
@@ -0,0 +1,197 @@
+        generateResults(messageContext, resultStatus, resultPayload);
+    }
+
+    /**
+     * Generate the results
+     *
+     * @param messageContext The message context that is processed by a handler in the handle method
+     * @param resultStatus   Result of the status (true/false)
+     * @param resultPayload  result payload create
+     */
+    private void generateResults(MessageContext messageContext, boolean resultStatus,
+                                 ResultPayloadCreate resultPayload) {
+        String response = FileConstants.START_TAG + resultStatus + FileConstants.END_TAG;
+        OMElement element;
+        try {
+            element = resultPayload.performSearchMessages(response);
+            resultPayload.preparePayload(messageContext, element);
+        } catch (XMLStreamException e) {
+            log.error(e.getMessage());
+            handleException(e.getMessage(), e, messageContext);
+        } catch (IOException e) {
+            log.error(e.getMessage());
+            handleException(e.getMessage(), e, messageContext);
+        } catch (JSONException e) {
+            log.error(e.getMessage());
+            handleException(e.getMessage(), e, messageContext);
+        }
+    }
+
+    /**
+     * Copy files
+     *
+     * @param source         Location of the file
+     * @param destination    new file location
+     * @param filePattern    pattern of the file
+     * @param messageContext The message context that is generated for processing the file
+     * @param opts           FileSystemOptions
+     * @return return a resultStatus
+     */
+    private boolean copyFile(String source, String destination, String filePattern,
+                             MessageContext messageContext, FileSystemOptions opts) {
+        boolean resultStatus = false;
+        StandardFileSystemManager manager = null;
+        try {
+            manager = FileConnectorUtils.getManager();
+            FileObject souFile = manager.resolveFile(source, opts);
+            FileObject destFile = manager.resolveFile(destination, opts);
+            if (StringUtils.isNotEmpty(filePattern)) {
+                FileObject[] children = souFile.getChildren();
+                for (FileObject child : children) {
+                    if (child.getType() == FileType.FILE) {
+                        copy(source, destination, filePattern, opts);
+                    } else if (child.getType() == FileType.FOLDER) {
+                        String newSource = source + File.separator + child.getName().getBaseName();
+                        copyFile(newSource, destination, filePattern, messageContext, opts);
+                    }
+                }
+                resultStatus = true;
+            } else {
+                if (souFile.exists()) {
+                    if (souFile.getType() == FileType.FILE) {
+                        InputStream fileIn = null;
+                        OutputStream fileOut = null;
+                        try {
```

Comment:
Could you see if we could bring this into a single try-catch block. It would look more readable

Reply:
Fixed in wso2-extensions/esb-connector-file#28